code/modules/antagonists/obsessed/obsessed.dm

```diff
@@ -0,0 +1,334 @@
+#define OBJECTIVE_STALK "spendtime"
+#define OBJECTIVE_PHOTOGRAPH "polaroid"
+#define OBJECTIVE_STEAL "heirloom"
+#define OBJECTIVE_JEALOUS "jealous"
+
+/datum/antagonist/obsessed
+	name = "Obsessed"
+	show_in_antagpanel = TRUE
+	antagpanel_category = "Other"
+	job_rank = ROLE_OBSESSED
+	show_name_in_check_antagonists = TRUE
+	roundend_category = "obsessed"
+	var/datum/mind/target
+	var/mob/living/carbon/human/human_target
+	var/total_time_stalking
+	var/time_spent_away
+	var/seen_alive = TRUE
+
+/datum/antagonist/obsessed/greet()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/creepalert.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	to_chat(owner, "<span class='userdanger'>You are the Obsessed!</span>")
+	to_chat(owner, "<B>The Voices have reached out to you, and are using you to complete their evil deeds.</B>")
+	to_chat(owner, "<B>You don't know their connection, but The Voices compel you to stalk [target], forcing them into a state of constant paranoia.</B>")
+	to_chat(owner, "<B>The Voices will retaliate if you fail to complete your tasks or spend too long away from your target.</B>")
+	to_chat(owner, "<span class='boldannounce'>This role does NOT enable you to otherwise surpass what's deemed creepy behavior per the rules.</span>")//ironic if you know the history of the antag
+	owner.current.client?.tgui_panel?.give_antagonist_popup("Obsession",
+		"Stalk [human_target.real_name] and force them into a constant state of paranoia.")
+
+/datum/antagonist/obsessed/on_gain()
+	find_target()
+	if(!target)
+		qdel(src)
+		return
+	//We need to find target first before calling parent here
+	. = ..()
+	forge_objectives()
+	RegisterSignal(owner.current, COMSIG_GLOB_MOB_DEATH, .proc/OnDeath)
+	RegisterSignal(owner.current, COMSIG_LIVING_REVIVE, .proc/OnRevival)
+	RegisterSignal(human_target, COMSIG_PARENT_QDELETING, .proc/TargetDeleted)
+	START_PROCESSING(SSprocessing, src)
+
+/datum/antagonist/obsessed/on_removal()
+	STOP_PROCESSING(SSprocessing, src)
+	UnregisterSignal(owner.current, COMSIG_GLOB_MOB_DEATH)
+	UnregisterSignal(owner.current, COMSIG_LIVING_REVIVE)
+	if(human_target)
+		UnregisterSignal(human_target, COMSIG_PARENT_QDELETING)
+		if(!seen_alive && human_target.stat != DEAD)
+			UnregisterSignal(human_target, COMSIG_LIVING_REVIVE)
+	to_chat(owner, "<span class='notice'>Your mind clears out!")
+	to_chat(owner, "<span class='userdanger'>You are no longer Obsessed!")
+	owner.current.client?.tgui_panel?.clear_antagonist_popup()
+	return ..()
+
+/datum/antagonist/obsessed/process(delta_time)
+	if(get_dist(get_turf(owner.current), get_turf(human_target)) > 7)//we're simply out of range
+		if(seen_alive)	//we know our target lives
+			time_spent_away += SSprocessing.wait * delta_time
```

Comment:
Why are you multiplying `delta_time` with `wait`? `delta_time` is already based on `wait`, no?